### PR TITLE
pykms: init at 20170719 and corresponding NixOS module

### PR DESCRIPTION
**Pull request by peterhoeg**
###### Motivation for this change

Have been running it here for testing purposes on 17.03 and 17.09 without problems.

While it works, I don't feel particularly proud of how the binaries are being wrapped as well as the handling of the DB creation script, so I'm very open to feedback on that.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/tools/networking/pykms/default.nix [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    owner  = "ThunderEX";
+    repo   = "py-kms";
+    rev    = "27355d88affd740330174a7c2bae9f50b9efce56";
+    sha256 = "0cpywj73jmyijjc5hs3b00argjsdwpqzmhawbxkx3mc2l4sgzc88";
+  };
+
+  propagatedBuildInputs = with python2Packages; [ argparse pytz ];
+
+  prePatch = ''
+    siteDir=$(toPythonPath $out)
```
Comment (FRidh) [inline pkgs/tools/networking/pykms/default.nix]: `siteDir=$out/${python.sitePackages}` ?

File: pkgs/tools/networking/pykms/default.nix [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+    ${python2Packages.python.interpreter} -m compileall $siteDir
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Windows KMS (Key Management Service) server written in Python";
+    homepage    = "https://forums.mydigitallife.info/threads/50234-Emulated-KMS-Servers-on-non-Windows-platforms/page2?p=840410&viewfull=1#post840410";
+    license     = licenses.mit;
+    maintainers = with maintainers; [ peterhoeg ];
+    platforms   = platforms.unix;
```
Comment (FRidh) [inline pkgs/tools/networking/pykms/default.nix]: Not needed

File: pkgs/tools/networking/pykms/default.nix (hunk 1 above)
Comment (peterhoeg) [reply pkgs/tools/networking/pykms/default.nix]: What's the difference?

Comment (peterhoeg) [reply pkgs/tools/networking/pykms/default.nix]: For the record, I have no issue at all with your version but I'm genuinely curious about the reason.

Comment (FRidh) [reply pkgs/tools/networking/pykms/default.nix]: I personally prefer to use Nix as much as possible. This is also more common I think.

**Review by akaWolf — CHANGES_REQUESTED** [review]
I would merge this 472e24429a47feec7dd7da681683f84819ee8597 commit into one of already existing.

**Comment by peterhoeg** [comment]
All, thanks for your input!

**Pull request merged**